loader.php

```diff
@@ -68,23 +68,40 @@ function( $asset_path ) { return 'build/' . $asset_path; },
  * @return string|null
  */
 function infer_base_url( string $path ) {
-	if ( strpos( $path, get_stylesheet_directory() ) === 0 ) {
-		return get_theme_file_uri( substr( $path, strlen( get_stylesheet_directory() ) ) );
+	$path = wp_normalize_path( $path );
+
+	$stylesheet_directory = wp_normalize_path( get_stylesheet_directory() );
+	if ( strpos( $path, $stylesheet_directory ) === 0 ) {
+		return get_theme_file_uri( substr( $path, strlen( $stylesheet_directory ) ) );
 	}
 
-	if ( strpos( $path, get_template_directory() ) === 0 ) {
-		return get_theme_file_uri( substr( $path, strlen( get_template_directory() ) ) );
+	$template_directory = wp_normalize_path( get_template_directory() );
+	if ( strpos( $path, $template_directory ) === 0 ) {
+		return get_theme_file_uri( substr( $path, strlen( $template_directory ) ) );
 	}
 
 	// Any path not known to exist within a theme is treated as a plugin path.
-	$plugin_path = plugin_dir_path( __FILE__ );
+	$plugin_path = get_plugin_basedir_path();
 	if ( strpos( $path, $plugin_path ) === 0 ) {
-		return plugin_dir_url( __FILE__ ) . substr( $path, strlen( $plugin_path ) );
+		return plugin_dir_url( __FILE__ ) . ltrim( substr( $path, strlen( $plugin_path ) ), '/' );
```

Comment:
FWIW, `untrailingslashit()` on `plugin_dir_url()` would probably be cleaner.

Reply:
No, because we do want a slash—but just a single one. I'll update this to use `str_len() + 1`, though.